### PR TITLE
Auth frontend

### PR DESCRIPTION
**Pull request by smacker**
I changed the token response to be compatible with our regular response format.

This PR doesn't cover proper serialization of responses, there are only a few new lines of code to create the same data shape for frontend. We can copy-past it from gitbase-web later or create some improved version.

I also introduced new `/api/me` endpoint. I think it belongs to auth service.

API js code is a little bit different (simpler) than what we did in other projects.
Apparently, we don't need all that checks if we follow correct types.

Implemented in separate PR:
- frontend router
- logout button

TODO:
- add configuration for AllowedOrigins (later in separate PR or never) https://github.com/src-d/lookout/issues/434
- improve serialization (later in separate PR) No special issue because it will be covered by one of BE tasks anyway

### CODE REVIEW

File: web/auth.go [hunk 1]
```diff
@@ -17,6 +17,38 @@ import (
...
+	if err != nil {
+		lg.RequestLog(r).Warn(err.Error())
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	w.Write(b)
+}
+
+func errorJSON(w http.ResponseWriter, r *http.Request, code int, errors ...error) {
```
Comment (carlosms) [inline web/auth.go]: Why is this defined, but then `http.Error` is used?

Comment (smacker) [reply web/auth.go]: just for testing. As I mentioned in PR description: "This PR doesn't cover proper serialization of responses"

File: web/http_server.go [hunk 2]
```diff
@@ -16,7 +16,9 @@ type HTTPServer struct {
 
 func NewHTTPServer(auth *Auth, static *Static) *HTTPServer {
 	corsOptions := cors.Options{
-		AllowedOrigins:   []string{"*"},
+		// TODO: make it customizable
+		// we can't pass "*" because it's incompatible with "credentials: include" request
+		AllowedOrigins:   []string{"http://127.0.0.1:3000"},
```
[the remark this answers is not in the document]
Comment (smacker) [reply web/http_server.go]: it's default dev port for create-react-app.

**Comment by smacker** [comment]
Router is required by the issue description written by you.

Do you suggest to add more `if`s for `/login`, `/unauthorized`, `/logout`?

File: web/http_server.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlosms) [reply web/http_server.go]: OK thanks. Then maybe we could add a comment saying why this is here, it's not so obvious that this is required for development.

**Comment by carlosms** [comment]
I'm not suggesting anything, and I'm not against using client-side routing. I was asking why is it needed.

Going through the alternatives in my mind I see I was not aware of the need for client-side routing because I was thinking on the way we defined the SSO OKR design doc, with separate endpoints for login and logout. Which in this case will probably complicate things too much, and it makes sense to keep it as a single page app with all the token logic in the same place.

On the other hand, if the goal of the frontend was to keep it as simple as possible, we could also skip routing and have the frontend work as the `/` path only, and use `/login` as api endpoint, just like this PR is doing right now.

Anyway I see the router already implemented in #433 so I guess there is no point in discussing if we need to implement it or not.

**Comment by carlosms** [comment]
Link to issue: #399

**Comment by smacker** [comment]
I don't mind dropping routing PR if everybody agrees to change original requirements and serve everything on `/` (except callback from github). You can review the new code and let me know if you think we don't need it.

**Comment by carlosms** [comment]
LGTM, we can discuss the need for a router in #433.

**Pull request merged**